scripts/tools/AddTaskToQueue.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2018 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoTaskQueue\scripts\tools;
+
+use common_report_Report as Report;
+use Zend\ServiceManager\ServiceLocatorAwareTrait;
+use Zend\ServiceManager\ServiceLocatorAwareInterface;
+use oat\oatbox\action\Action;
+use oat\taoTaskQueue\model\QueueDispatcherInterface;
+
+/**
+ * Class AddTaskToQueue
+ *
+ * Action to add task to the task queue.
+ *
+ * Run example:
+ * ```
+ * sudo -u www-data php index.php '\oat\taoTaskQueue\scripts\tools\AddTaskToQueue' '\Task\To\Be\Run' param1 param2
+ * ```
+ *
+ * First parameter is the task class including namespace. Should be instance of oat\oatbox\action\Action interface
+ * Further parameters are parameters to be passed to the task's __invoke() function at time of launch.
+ *
+ * @package oat\taoTaskQueue\scripts\tools
+ * @author Aleh Hutnikau, <hutnikau@1pt.com>
+ */
+class AddTaskToQueue implements Action, ServiceLocatorAwareInterface
+{
+    use ServiceLocatorAwareTrait;
+
+    public function __invoke($params)
+    {
+        $action = array_shift($params);
+
+        if (!class_exists($action)) {
+            return Report::createFailure('Action class does not exist');
+        }
+
+        if (!is_subclass_of($action, Action::class)) {
+            return Report::createFailure('Action in not instance of ' . Action::class);
+        }
+
+        $actionInstance = new $action;
+        /** @var QueueDispatcherInterface $queueDispatcher */
+        $queueDispatcher = $this->getServiceLocator()->get(QueueDispatcherInterface::SERVICE_ID);
+        $queueDispatcher->createTask($actionInstance, $params, $action);
```

Comment:
What's the reason for this new feature? Is it a solution for only some custom issue or you want this feature being used globally for every type of `Action`? Because this way of handling of parameters won't work for all actions. If the given action expecting an array with keys, than the given task will fail during processing, like `\oat\taoDeliveryRdf\model\tasks\CompileDelivery Line 56`. So this solution only works if the task expect a numeric indexed array of parameters, of course in the proper order. If we want to use this command globally somehow this command should take care of those type of parameters and pre-validate them, so we can be notified in advance if there is some problem with the parameters. 

If this is only used for a specific `Action` class (for a specific client) which always expect a numeric indexed array as parameter, than maybe this code should not be put here.

Reply:
The main reason is to be able to add task to the queue from scheduler.
see https://github.com/oat-sa/extension-tao-act/pull/952/files#diff-c6868ad8dbc7f5e2a4993d9ac58f0713R66

Also it may be useful to me able to add tasks to queue from cli.

Not sure that I understand what do you mean about params as an array. Why can't we pass array as the first parameter? Like:
```
$addTaskToQueueActionInstance(['action/to/do', ['param1', 'param2']]);
```

Reply:
Yes, there might be a misunderstanding here. Passing parameters in array, of course, it's fine, I'm talking about what if the action expects a param array with specified keys, like the action what I mentioned above. So if I create a new `CompileDelivery` using this new command from CLI, the task will fail during process because `CompileDelivery` expects a `test` and `deliveryClass` keys in the params array which are not possible to provide through CLI.

Reply:
Running this action fom CLI was a use-case that I didn't even take into account, as it was only intended to be used from the scheduler. But if we take into account that CompileDelivery cannot be launched from CLI anyway, I don't think this could be considered a regression

Reply:
I added unit tests to check that this action works with both instance of `oat\oatbox\action\Action` and instance of `oat\oatbox\extension\script\ScriptAction` (parametrized one).
Also you can test launch of actions from cli:
```
php index.php '\oat\taoTaskQueue\scripts\tools\AddTaskToQueue' '\oat\taoTaskQueue\test\scripts\tools\mock\ScriptActionMock' --param1 foo --param2 bar
php index.php '\oat\taoTaskQueue\scripts\tools\AddTaskToQueue' '\oat\taoTaskQueue\test\scripts\tools\mock\ActionMock' param1 param2
```